pkg/standalone-utils/test/VaultActions.test.ts

```diff
@@ -1161,4 +1190,324 @@ describe('VaultActions', function () {
+          // So external token balances of DAI/MKR should be unchanged, and SNX should equal token out from swap
+          const receipt = await (
+            await relayer.connect(user).multicall([
+              encodeExitPool(vault, relayerLibrary, tokens, {
+                poolKind: PoolKind.WEIGHTED,
+                poolId: poolIdA,
+                userData: WeightedPoolEncoder.exitExactBPTInForTokensOut(amountInBPT),
+                toInternalBalance: true,
+                outputReferences: {
+                  DAI: toChainedReference(0),
+                  MKR: toChainedReference(1),
+                },
+                sender,
+                recipient: relayer.address,
+              }),
+              encodeBatchSwap({
+                swaps: [{ poolId: poolIdB, tokenIn: tokens.MKR, tokenOut: tokens.SNX, amount: toChainedReference(1) }],
+                outputReferences: {
+                  SNX: toChainedReference(1),
+                },
+                sender: relayer.address,
+                recipient: TypesConverter.toAddress(sender),
+                useInternalBalance: true,
+              }),
+              encodeManageUserBalance({
+                ops: [
+                  {
+                    kind: UserBalanceOpKind.WithdrawInternal,
+                    asset: tokens.SNX.address,
+                    amount: toChainedReference(1),
+                    sender,
+                  },
+                ],
+              }),
+            ])
+          ).wait();
+
+          let daiAmountOut = Zero;
+
+          const daiTransfer = expectEvent.inIndirectReceipt(
+            receipt,
+            vault.instance.interface,
+            'InternalBalanceChanged',
+            {
+              user: TypesConverter.toAddress(relayer),
+              token: tokens.DAI.address,
+            }
+          );
+
+          daiAmountOut = daiTransfer.args.delta;
+
+          expectEvent.inIndirectReceipt(receipt, vault.instance.interface, 'InternalBalanceChanged', {
+            user: TypesConverter.toAddress(relayer),
+            token: tokens.MKR.address,
+          });
+
+          const snxTransfer = expectEvent.inIndirectReceipt(
+            receipt,
+            vault.instance.interface,
+            'InternalBalanceChanged',
+            {
+              user: TypesConverter.toAddress(sender),
+              token: tokens.SNX.address,
+            }
+          );
+
+          const snxAmountWithdrawn = snxTransfer.args.delta;
+
+          const {
+            args: { amountOut: amountOutSNX },
+          } = expectEvent.inIndirectReceipt(receipt, vault.instance.interface, 'Swap', { poolId: poolIdB });
+
+          await expectChainedReferenceContents(relayer, toChainedReference(0), daiAmountOut);
+          expect(snxAmountWithdrawn).to.eq(amountOutSNX);
+
+          // SNX should be in recipient's account.
+          expect(await tokens.SNX.balanceOf(recipient)).to.eq(amountOutSNX);
```

Comment:
maybe checking the external transfer event would be safer using `expectTransferEvent`?
We are not checking whether the recipient had SNX before the transfers -- by any means, this check should be correct anyways.